engine/src/main/java/com/datatorrent/stram/plan/physical/PhysicalPlan.java

```diff
@@ -979,6 +998,14 @@ public void deployChanges() {
     this.undeployOpers.removeAll(newOpers.keySet());
```

Comment:
@tweise The getDependents also includes the operators in the argument set, since they are new we don't want to undeploy them. So it should be this

Set<PTOperator> ndeps = getDependents(newOpers.keySet());
ndeps.removeAll(newOpers.keySet());
this.undeployOpers.addAll(ndeps);

As you pointed out ndeps doesn't have to be added to deployOpers as it is already being done in the next line.

I had a line in there 
ndeps.removeAll(this.undeployOpers);
to ensure that operators that are already scheduled to be undeployed (removed completely) don't get re-deployed because of this but on further thought this probably is not required because if some operators are already going to be undeployed they will not be in the dependency chain for newly deployable operators, is that assumption valid?